Applications/Spire/Source/Ui/ClosedFilterPanel.cpp

```diff
@@ -0,0 +1,216 @@
+#include "Spire/Ui/CustomQtVariants.hpp"
+#include "Spire/Ui/FilterPanel.hpp"
+#include "Spire/Ui/ListItem.hpp"
+#include "Spire/Ui/ListValueModel.hpp"
+#include "Spire/Ui/ListView.hpp"
+#include "Spire/Ui/ScrollBar.hpp"
+#include "Spire/Ui/ScrollBox.hpp"
+#include "Spire/Ui/ScrollableListBox.hpp"
+
+using namespace boost;
+using namespace boost::signals2;
+using namespace Spire;
+using namespace Spire::Styles;
+
+namespace {
+  auto LIST_ITEM_STYLE() {
+    auto style = StyleSheet();
+    style.get(Any()).
+      set(BackgroundColor(QColor(0xFFFFFF))).
+      set(border_size(0)).
+      set(PaddingLeft(scale_width(8))).
+      set(PaddingRight(scale_width(10))).
+      set(vertical_padding(scale_height(5)));
+    return style;
+  }
+
+  auto LIST_VIEW_STYLE(StyleSheet style) {
+    style.get(Any()).
+      set(EdgeNavigation::CONTAIN);
+    return style;
+  }
+
+  void invalidate_descendants(QWidget& widget) {
+    for(auto child : widget.children()) {
+      if(!child->isWidgetType()) {
+        continue;
+      }
+      auto& widget = *static_cast<QWidget*>(child);
+      invalidate_descendants(widget);
+      widget.updateGeometry();
+      if(widget.layout()) {
+        widget.layout()->invalidate();
+      }
+    }
+  }
+}
+
+class BooleanListValueModel : public BooleanModel {
+  public:
+    using Type = BooleanModel::Type;
+
+    explicit BooleanListValueModel(std::shared_ptr<ListValueModel> source)
+      : m_source(std::move(source)),
+        m_source_connection(m_source->connect_current_signal(
+          std::bind_front(&BooleanListValueModel::on_current, this))) {}
+
+    const Type& get_current() const override {
+      return std::any_cast<const Type&>(m_source->get_current());
+    }
+
+    QValidator::State set_current(const Type& value) override {
+      return m_source->set_current(value);
+    }
+
+    connection connect_current_signal(
+        const typename CurrentSignal::slot_type& slot) const override {
+      return m_current_signal.connect(slot);
+    }
+
+    void on_current(const std::any& current) {
+      m_current_signal(std::any_cast<const Type&>(current));
+    }
+
+  private:
+    mutable CurrentSignal m_current_signal;
+    std::shared_ptr<ListValueModel> m_source;
+    scoped_connection m_source_connection;
+};
+
+ClosedFilterPanel::ClosedFilterPanel(std::shared_ptr<TableModel> model,
+    QString title, QWidget& parent)
+    : QWidget(&parent),
+      m_model(std::move(model)),
+      m_submission(std::make_shared<ArrayListModel>()) {
+  for(auto i = 0; i < m_model->get_row_size(); ++i) {
+    if(m_model->get<bool>(i, 1)) {
+      m_submission->push(m_model->at(i, 0));
+    }
+  }
+  m_list_view = new ListView(
+    std::make_shared<ColumnViewListModel>(m_model, 1),
+    [=] (const auto& model, int index) {
+      auto boolean_model = std::make_shared<BooleanListValueModel>(
+        std::make_shared<ListValueModel>(model, index));
+      auto connection = boolean_model->connect_current_signal(
```

Comment:
This connection most likely should be removed. Listen to ```m_model``` or the ```ColumnViewListModel``` which already do the work of managing connections, instead of having to do all this connection management. You can also remove ```m_model_items``` with that change.